### PR TITLE
[ablation] Support for module ablation

### PR DESCRIPTION
**Pull request by ssheikholeslami**
Adds support for ablation of modules wrapped between concatenation (mixed) layers, e.g., Inception modules.
Refactors the `loco` ablator by moving most of the model and dataset manipulation code to the `utils` module.

### CODE REVIEW

**Review by moritzmeister — APPROVED** [review]

**Comment by ssheikholeslami** [comment]
I'll be pushing some fixes and improvements soon.

**Comment by ssheikholeslami** [comment]
Tested on Hopsworks 1.3.0 (TF 1.15.0).

File: maggy/ablation/ablationstudy.py [hunk 1]
```diff
@@ -249,6 +254,11 @@ def add_custom_model_generator(self, custom_model_generator, model_identifier):
         """
         self.custom_model_generators.append((custom_model_generator, model_identifier))
 
+    def add_module(self, starting_layer, ending_layer):
+        """
```
Comment (moritzmeister) [inline maggy/ablation/ablationstudy.py:258]: do you want to leave the docstring like this?

File: maggy/ablation/utils/functionalmodel.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+
+
+def get_list_of_layer_names(config_dict):
+    """
+        Return a list of layer names.
```
Comment (moritzmeister) [inline maggy/ablation/utils/functionalmodel.py:20]: no need for the indent

**Review by moritzmeister — APPROVED** [review]

**Comment by moritzmeister** [comment]
@ssheikholeslami do you want to address my comments?

**Comment by ssheikholeslami** [comment]
Will wait for new hopsworks release with TF 2.2 to make changes.

**Comment by moritzmeister** [comment]
outdated

**Pull request closed without merging**